### DISCUSSION TITLE
Designing our IRIs

### DISCUSSION THREAD

**Discussion by cspayne** [crystalyragui/MARC2RDA#472] [General]
[Minting IRIs Discussion](https://github.com/uwlib-cams/MARC2RDA/discussions/450)

I have just pushed an update to the code that helps to ensure our minted IRIs remain unique by basing them off of the control number in 001. 
I have also restructured the IRIs slightly, based heavily off of how [marc2bibframe designed theirs](https://github.com/lcnetdev/marc2bibframe2?tab=readme-ov-file#converter-design).

IRIs are structured as follows:

For the main WEM stack: `[BASE]/[control number]wor`, `[BASE]/[control number]exp`, and `[BASE]/[control number]man` 
e.g. `http://marc2rda.edu/fake/99162596149401452wor`

For opaque IRIs, they are structured similarly, but with an additional identifier created by the transform added as a hash URI.
`[BASE]/[control number][type]#[generated id]`. The type may be `relWor`, `origMan`, `ite`, `nom`, `age`, `pla`, `tim`, or `con` based on what entity the IRI is minted for. 
e.g. `http://marc2rda.edu/fake/99115297060001452ite#d22e8053`

For IRIs where sources are provided and we want to allow for some automatic de-duplication, the structure is `[BASE]/[source]/[type]#[access point]`. The control number is not used so that de-duplication can occur across records. 
e.g. `http://marc2rda.edu/fake/star/con#restrictionsunspecified` and `http://marc2rda.edu/fake/lcsh/age#unitedstatesdepartmentofagricultureofficeofthesecretary`

Any input or thoughts on how we structure and mint IRIs is welcome and can go in this discussion.

**Comment by CECSpecialistI** [comment]
I’m just not sure about this at all. I’ve asked our head of ITS-DS to make sure we don’t have any policies or guidelines here at UW that would require us to shape IRIs in a certain way. If we don’t, we’re just going to have to come up with the best solution we can. Soon.
